src/app/shared/github/github.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { EMPTY, Observable, of } from 'rxjs';
+import { delay, expand, map, toArray } from 'rxjs/operators';
+
+import { environment } from '~/environments/environment';
+import { GithubTag } from '~/app/shared/github/github.interfaces';
+import { tags } from './mocks/tags';
+import { compare } from './mocks/compare';
+import { associatedPullReqeusts } from './mocks/associated-pull-request';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GithubService {
+  useMocks: boolean = false;
+  constructor(private http: HttpClient) {}
+
+  // TODO: GET ALL TAGS
+  getTags(page = 1): Observable<any> {
+    if (this.useMocks) {
+      return of(tags);
+    }
+    const options = {
+      headers: new HttpHeaders({
+        Authorization: 'token ' + environment.oauth.githubToken1 + environment.oauth.githubToken2,
```

Comment:
Consider getting the HttpHeaders from a private function (using the same code at 3 places in this file)